numpy/core/src/multiarray/datetime.c

```diff
@@ -3775,7 +3781,18 @@ time_to_time_resolve_descriptors(
     meta2 = get_datetime_metadata_from_dtype(loop_descrs[1]);
     assert(meta2 != NULL);
 
-    if (meta1->base == meta2->base && meta1->num == meta2->num) {
+    if ((meta1->base == meta2->base && meta1->num == meta2->num) ||
+        // handle some common metric prefix conversions
+        // 1000 fold conversions
+        ((meta2->base >= 7) && (meta1->base - meta2->base == 1)
+          && ((meta1->num / meta2->num) == 1000)) ||
+        // 10^6 fold conversions
+        ((meta2->base >= 7) && (meta1->base - meta2->base == 2)
+          && ((meta1->num / meta2->num) == 1000000)) ||
+        // 10^9 fold conversions
+        ((meta2->base >= 7) && (meta1->base - meta2->base == 3)
+          && ((meta1->num / meta2->num) == 1000000000))
+        )  {
```

Comment:
I would prefer the numpy-style over-indent (if you prefer the closing `)` on the new-line, I can live with it :).

Reply:
I tried to change the whitespace--not sure if I got it perfect yet.

Reply:
Oh, most code in NumPy indents the second row of if statements (and function calls) by 8 spaces, to offset it from the code block below.  That was what I meant.  I guess that is starting to be less common these days in general, though...

Reply:
Like this?

```diff
diff --git a/numpy/core/src/multiarray/datetime.c b/numpy/core/src/multiarray/datetime.c
index 59155e059..7454bf1a7 100644
--- a/numpy/core/src/multiarray/datetime.c
+++ b/numpy/core/src/multiarray/datetime.c
@@ -3782,16 +3782,16 @@ time_to_time_resolve_descriptors(
     assert(meta2 != NULL);
 
     if ((meta1->base == meta2->base && meta1->num == meta2->num) ||
-        // handle some common metric prefix conversions
-        // 1000 fold conversions
-        ((meta2->base >= 7) && (meta1->base - meta2->base == 1)
-        && ((meta1->num / meta2->num) == 1000)) ||
-        // 10^6 fold conversions
-        ((meta2->base >= 7) && (meta1->base - meta2->base == 2)
-        && ((meta1->num / meta2->num) == 1000000)) ||
-        // 10^9 fold conversions
-        ((meta2->base >= 7) && (meta1->base - meta2->base == 3)
-        && ((meta1->num / meta2->num) == 1000000000))) {
+            // handle some common metric prefix conversions
+            // 1000 fold conversions
+            ((meta2->base >= 7) && (meta1->base - meta2->base == 1)
+            && ((meta1->num / meta2->num) == 1000)) ||
+            // 10^6 fold conversions
+            ((meta2->base >= 7) && (meta1->base - meta2->base == 2)
+            && ((meta1->num / meta2->num) == 1000000)) ||
+            // 10^9 fold conversions
+            ((meta2->base >= 7) && (meta1->base - meta2->base == 3)
+            && ((meta1->num / meta2->num) == 1000000000))) {
         if (byteorder_may_allow_view) {
             return NPY_NO_CASTING | byteorder_may_allow_view;
         }

```

Reply:
Yeah.  The additional space you had for the `&&` was good IMO.